eclipse-sirius/sirius-web
# [3429] Add diagram filter view in diagram panel

Bug: https://github.com/eclipse-sirius/sirius-web/issues/3429

# Pull request template

## General purpose
What is the main goal of this pull request?

## Project management

## Architectural decision records (ADR)

## Dependencies

## Frontend

This section is not relevant if your contribution does not come with changes to the frontend.

### General purpose

### Typing
We need to improve the typing of our code, as such, we require every contribution to come with proper TypeScript typing for both changes contributing new files and those modifying existing files.
Please ensure that the following statements are true for each file created or modified (this may require you to improve code outside of your contribution).

- Hooks are properly typed:

## Backend

This section is not relevant if your contribution does not come with changes to the backend.

### General purpose

### Architecture

## Review

### How to test this PR?
1. Start a Sirius Web application (new architecture)
2. Create a studio
3. Open the diagram representation inside _DomainNewModel_
4. Click on the _diagram filters_ tool in the diagram panel
5. Select some nodes and apply tools on them
6. The diagram should be updated to reflect the tools you applied
